Eliminate preprocessing requirement with C++17

GravisZro opened the pull request:
With only minor alterations, I was able to eliminate all preprocessing requirements.  It's still easy to read, so I do not believe this is reason to object.

olegendo commented:
This is a funny one. Having the s-expressions inlined in the C++ code was the whole fun part of this. Of course it can be done as plain C++. Somebody also asked to pull out the data in some JSON format and separate it from the HTML page generation code. Not sure how to go about this one. Unfortunately I like s-expressions too much ;)

GravisZro commented:
That's fine. The reason I took them out was because it required additional compilation steps.

Anyway, I've moved far away since these PRs. I've added more data, corrected some, and added features using CSS and data processing/post-processing.
* output: CSS only, no JavaScript
* output: light/dark mode color schemes (automatically selected by browser)
* output: sticky table headers
* output: mouse over for acronyms
* gen/output: opcodes subsection highlighting with mouse over details
* data/output: actual instruction names with highlighted mnemonic characters
* gen/data: formatted example assembly from the docs (WIP)
* data: fixed/optimized the SVG images (didn't show up properly in dark mode)
* gen/data/output: links to the pages and datasheets the info is from (WIP)
* gen/data/output: valid HTML (WIP) (because [OMG](https://validator.w3.org/check?uri=http%3A%2F%2Fwww.shared-ptr.com%2Fsh_insns.html&charset=%28detect+automatically%29&doctype=Inline&group=0))
* gen: data is grouped
* gen: groups are cross referenced to identify missing data (very handy as a lot was missing!)

The intent is to make a more accessible version of the information in the datasheets while being able to reduce the information provided to only the processor(s)/chips you are interested in.  At this point, it's more like a database with cross referenced tables and I'm considering actually making a sqlite DB.

If you're curious about what I've done then here's [the latest generated HTML](https://saturnopensdk.github.io/). (CPU drop down menu is currently non-functional)

olegendo commented:
Your version looks nice!
